### PR TITLE
index: handle extractor ping failure properly

### PR DESCRIPTION
**Pull request by robn**
Status code can be trampled inside `http_read_response` in certain failure conditions, so it being non-zero is not enough of an indicator of success.

```
019-09-19T15:05:15.553914-04:00 imap99 sloti99d1t03/squatter[879305]: indexing mailbox user.robntest@fastmail.fm...
2019-09-19T15:05:15.554285-04:00 imap99 sloti99d1t03/squatter[879305]: extractor_connect(http://localproxy:9871/extract)
2019-09-19T15:05:15.554291-04:00 imap99 sloti99d1t03/squatter[879305]: extractor_connect: ping existing connection
2019-09-19T15:05:15.554350-04:00 imap99 sloti99d1t03/squatter[879305]: extractor_ping: HEAD /extract: failed (-1964266953): Unable to read status-line from backend
2019-09-19T15:05:15.554354-04:00 imap99 sloti99d1t03/squatter[879305]: extractor_disconnect(0x24f37f0)
2019-09-19T15:05:15.554373-04:00 imap99 sloti99d1t03/squatter[879305]: extractor_connect: reconnect to localproxy:9871/noauth
2019-09-19T15:05:15.564125-04:00 imap99 sloti99d1t03/squatter[879305]: read & parse headers
2019-09-19T15:05:15.564171-04:00 imap99 sloti99d1t03/squatter[879305]: http_read_body(flags=0x1, framing=0)
2019-09-19T15:05:15.564427-04:00 imap99 sloti99d1t03/squatter[879305]: extract_attachment: GET /extract/715cb01fe553bc8b195ecd870244094c6f56a8e0: got status 200
2019-09-19T15:05:15.636321-04:00 imap99 sloti99d1t03/squatter[879305]: Xapian committed 1 updates in 0.014942 sec
```

### CODE REVIEW

File: imap/index.c [hunk 1]
```diff
@@ -5226,18 +5227,25 @@ static int ping(struct backend *s,
...
+        if (r) {
             break;
         }
     } while (statuscode < 200);
 
-    syslog(LOG_DEBUG, "extractor_ping: HEAD %s: got status %u",
-           ctx->path, statuscode);
+    if (r) {
+        syslog(LOG_DEBUG, "extractor_ping: HEAD %s: failed (%d): %s",
+               ctx->path, r, errstr);
```
Comment (elliefm) [inline imap/index.c]: instead of `%d` and `r`, I'd suggest `%s` and `error_message(r)` is more useful to have in the log

Comment (ksmurchison) [reply imap/index.c]: I agree with ellie.  Otherwise, this should be good

Comment (robn) [reply imap/index.c]: Fixed, and thanks. The more you know!

**Comment by robn** [comment]
@elliefm wanna give this another glance and (hopefully) 👍 and merge? Thanks!

**Comment by elliefm** [comment]
Merged, thanks!

**Pull request merged**